lib/harness/action_subscriber.rb

```diff
@@ -3,43 +3,61 @@
 require "harness"
 require "active_support"
 
-::ActiveSupport::Notifications.subscribe "message_acked.action_subscriber" do |*args|
-  event = ::ActiveSupport::Notifications::Event.new(*args)
-  event_name = event.payload[:queue].gsub(".", "-")
-  ::Harness.increment "action_subscriber.messages_acked.#{event_name}"
-  ::Harness.timing "action_subscriber.ack_duration.#{event_name}", event.duration
-end
+module Harness
+  module ActionSubscriber
+    UNBLOCKED_METRIC = ["action_subscriber", ENV["SERVICE_NAME"], "connection", "unblocked"].reject(&:nil?).join(".").freeze
 
-::ActiveSupport::Notifications.subscribe "message_nacked.action_subscriber" do |*args|
-  event = ::ActiveSupport::Notifications::Event.new(*args)
-  event_name = event.payload[:queue].gsub(".", "-")
-  ::Harness.increment "action_subscriber.messages_nacked.#{event_name}"
-  ::Harness.timing "action_subscriber.nack_duration.#{event_name}", event.duration
-end
+    ::ActiveSupport::Notifications.subscribe "connection_blocked.action_subscriber" do |_, _, _, _, params|
+      reason = params.fetch(:reason)
+      blocked_metric = ["action_subscriber", ENV["SERVICE_NAME"], "connection", "blocked", reason.gsub(/\W/, "_")].
```

Comment:
Is there a potential for `reason` to be `nil`?

Reply:
Giving you a :shipit: because I don't think there is, but I did want to bring it up in case you know of something that we're overlooking.

Reply:
It could potentially happen. I haven't seen it happen with any blocking events I've triggered. If it did though, the `nil` would be filtered out and we'd end up with `aciton_subscriber.app_name.connection.blocked` without a reason.

Reply:
But `reason.gsub(/\W/, "_")` would fail.